packages/esm-patient-registration-app/src/patient-registration/input/basic-input/input/input.test.tsx

```diff
@@ -34,33 +34,70 @@ describe.skip('number input', () => {
   });
 });
 
-describe.skip('text input', () => {
+describe('text input', () => {
   const setupInput = async () => {
     render(
-      <Formik initialValues={{ text: '' }} onSubmit={null}>
+      <Formik initialValues={{ text: '' }} onSubmit={() => {}}>
         <Form>
-          <Input id="text" labelText="Text" name="text" placeholder="Enter text" />
+          <Input
+            id="text"
+            labelText="Text"
+            name="text"
+            placeholder="Enter text"
+            required
+            checkWarning={(value) => {
+              if (value.length > 5) {
+                return 'name should be of 5 char';
+              }
+            }}
+          />
         </Form>
       </Formik>,
     );
-    return screen.getByLabelText('text') as HTMLInputElement;
+    return screen.getByLabelText('Text') as HTMLInputElement;
   };
 
   it('exists', async () => {
     const input = await setupInput();
     expect(input.type).toEqual('text');
   });
 
-  it('can input data', async () => {
+  it('can input valid data without warning', async () => {
     const user = userEvent.setup();
 
     const input = await setupInput();
-    const expected = 'Some text';
+    const expected = 'text';
 
     await user.type(input, expected);
     await user.tab();
 
     expect(input.value).toEqual(expected);
+    expect(screen.queryByText('name should be of 5 char')).not.toBeInTheDocument();
+  });
+
+  it('should show a warning when the invalid input is entered', async () => {
+    const user = userEvent.setup();
+
+    const input = await setupInput();
+    const expected = 'Hello World';
```

Comment:
Let's call this `userInput` because that's the part that user inputs. @ayush-AI 

```suggestion
    const userInput = 'Hello World';
```

Reply:
@ayush-AI Did you resolve this?

Reply:
@anjula-sack Now it is resolved.